test/integration/test_display_callback.py

```diff
@@ -311,3 +317,95 @@ def test_module_level_no_log(executor, playbook):
+        playbook=os.path.join(private_data_dir, 'fake_playbook.yml')
+    )
+
+    executor.run()
+    stdout = executor.stdout.read()
+    assert "ERROR! the playbook:" in stdout
+    assert "could not be found" in stdout
+
+
+def test_output_when_given_non_playbook_script(private_data_dir):
+    # As shown in the following pull request:
+    #
+    #   https://github.com/ansible/ansible-runner/pull/256
+    #
+    # This ports some functionality that previously lived in awx and allows raw
+    # lines of stdout to be treated as event lines.
+    #
+    # As mentioned in the pull request as well, there were no specs added, and
+    # this is a retro-active test based on the sample repo provided in the PR:
+    #
+    #   https://github.com/AlanCoding/ansible-runner-examples/tree/master/non_playbook/sleep_with_writes
+
+    with open(os.path.join(private_data_dir, "args"), 'w') as args_file:
+        args_file.write("bash sleep_and_write.sh\n")
+    with open(os.path.join(private_data_dir, "sleep_and_write.sh"), 'w') as script_file:
+        script_file.write("echo 'hi world'\nsleep 0.5\necho 'goodbye world'\n")
+
+    # Update the settings to make this test a bit faster :)
+    os.mkdir(os.path.join(private_data_dir, "env"))
+    with open(os.path.join(private_data_dir, "env", "settings"), 'w') as settings_file:
+        settings_file.write("pexpect_timeout: 0.2")
+
+    executor = init_runner(
+        private_data_dir=private_data_dir,
+        inventory="localhost ansible_connection=local",
+        envvars={"ANSIBLE_DEPRECATION_WARNINGS": "False"}
+    )
+
+    executor.run()
+    stdout = executor.stdout.readlines()
+    assert stdout[0].strip() == "hi world"
+    assert stdout[1].strip() == "goodbye world"
+
+    events = list(executor.events)
+
+    assert len(events) == 2
+    assert events[0]['event'] == 'verbose'
+    assert events[0]['stdout'] == 'hi world'
+    assert events[1]['event'] == 'verbose'
+    assert events[1]['stdout'] == 'goodbye world'
+
+
+@pytest.mark.parametrize('json_mode', [True])
+@pytest.mark.parametrize('playbook', [
+{'listvars.yml': '''
+- name: List Variables
+  connection: local
+  hosts: all
+  tasks:
+    - name: Display all variables/facts known for a host
+      debug:
+        var: hostvars[inventory_hostname]
+'''},  # noqa
+])
+def test_large_stdout_parsing_when_using_json_output(executor, playbook, json_mode):
+    # When the json flag is used, it is possible to output more data than
+    # pexpect's maxread default of 2000 characters.  As a result, if not
+    # handled properly, the stdout can end up being corrupted with partial
+    # non-event matches with raw "non-json" lines being intermixed with json
+    # ones.
+    #
+    # This tests to confirm we don't polute the stdout output with non-json
+    # lines when a single event has a lot of output.
+    executor.run()
+    assert all(line[0] == "{" for line in executor.stdout.readlines())
```

Comment:
I'm also somewhat unclear what this is looking for. Like, it's printing a dictionary, but only the first line needs to start with "{"

Reply:
When working properly, the JSON formatted output should print one JSON object per line, so this was a "low tech" way of confirming that was the case.  It is possible that I also wanted to avoid parsing JSON because prior to the fix, it would cause the test suite to stop abruptly, but I can't recall.

However, I also realize this test is brittle, so I am fine with it changing or being removed.